app/src/main/java/ru/yandex/practicum/contacts/presentation/filter/model/FilterContactTypeUi.java

```diff
@@ -20,6 +22,11 @@ public boolean isSelected() {
         return selected;
     }
 
+    @Override
+    public boolean theSameAs(FilterContactTypeUi obj) {
+        return this.hashCode() == obj.hashCode();
```

Comment:
```suggestion
        return this.getContactType() == obj.getContactType();
```
Здесь тоже нужно поправить, сравнивать по возможности нужно поля объекта, а не только хэшкоды